airbyte-integrations/connectors/source-hubspot/source_hubspot/spec.json

```
@@ -15,16 +15,77 @@
       },
       "credentials": {
         "type": "object",
-        "title": "api key",
-        "required": ["api_key"],
-        "properties": {
-          "api_key": {
-            "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
-            "type": "string",
-            "airbyte_secret": true
+        "oneOf": [
+          {
+            "type": "object",
+            "title": "api key",
+            "required": ["api_key", "credentials_title"],
+            "properties": {
+              "credentials_title": {
+                "type": "string",
+                "const": "API Key Credentials",
+                "enum": ["API Key Credentials"],
+                "default": "API Key Credentials",
+                "order": 0
+              },
+              "api_key": {
+                "description": "Hubspot API Key. See our <a href=\"https://docs.airbyte.io/integrations/sources/hubspot\">docs</a> if you need help finding this key.",
+                "type": "string",
+                "airbyte_secret": true
+              }
+            }
+          },
+          {
+            "type": "object",
+            "title": "oauth",
```

The persona using this auth mechanism will often be non-programmers who don't know what oauth is, so i'm making this suggestion to make it more obvious to them


```suggestion
            "title": "Authenticate via Hubspot (Oauth)",
```